src/swap/handle_swap_sign_transaction.c

```diff
@@ -0,0 +1,137 @@
+
+bool swap_copy_transaction_parameters(create_transaction_parameters_t* params) {
+    PRINTF("Inside swap_copy_transaction_parameters %s\n", params->destination_address);
+
+    if (params->destination_address == NULL) {
+        PRINTF("Destination address expected\n");
+        return false;
+    }
+
+    if (strlen(params->destination_address) != (ADDRESS_LEN * 2)) {
+        PRINTF("Destination address wrong length\n");
+        return false;
+    }
+
+    if (params->amount == NULL) {
+        PRINTF("Amount expected\n");
+        return false;
+    }
+
+    // first copy parameters to stack, and then to global data.
+    // We need this "trick" as the input data position can overlap with app globals
+    // and also because we want to memset the whole bss segment as it is not done
+    // when an app is called as a lib.
+    // This is necessary as many part of the code expect bss variables to
+    // initialized at 0.
+    swap_validated_t swap_validated;
+    memset(&swap_validated, 0, sizeof(swap_validated));
+
+    // Save recipient as an uppercase string
+    for (int i = 0; i < ADDRESS_LEN * 2; i++) {
+        if (params->destination_address[i] >= 'a' && params->destination_address[i] <= 'z') {
+            swap_validated.recipient[i] = params->destination_address[i] - 'a' + 'A';
+        } else {
+            swap_validated.recipient[i] = params->destination_address[i];
+        }
+    }
+
+    // Save amount
+    if (params->amount_length > sizeof(swap_validated.amount)) {
+        PRINTF("Amount too big\n");
+        return false;
+    } else {
+        // Convert params->amount to uint64_t
+        swap_validated.amount = 0;
+        memcpy(((uint8_t*) &swap_validated.amount) + sizeof(swap_validated.amount) -
+                   params->amount_length,
+               params->amount,
+               params->amount_length);
+        swap_validated.amount = __builtin_bswap64(swap_validated.amount);
+    }
+
+    // Save fee
+    if (params->fee_amount_length > sizeof(swap_validated.fee)) {
+        PRINTF("Fee too big\n");
+        return false;
+    } else {
+        // Convert params->fee_amount to uint64_t
+        swap_validated.fee = 0;
+        memcpy(((uint8_t*) &swap_validated.fee) + sizeof(swap_validated.fee) -
+                   params->fee_amount_length,
+               params->fee_amount,
+               params->fee_amount_length);
+        swap_validated.fee = __builtin_bswap64(swap_validated.fee);
+    }
+
+    swap_validated.initialized = true;
+
+    // Full reset the global variables
+    os_explicit_zero_BSS_segment();
+
+    // Commit from stack to global data, params becomes tainted but we won't access it anymore
+    memcpy(&G_swap_validated, &swap_validated, sizeof(swap_validated));
+
+    PRINTF("Out of swap_copy_transaction_parameters\n");
+
+    return true;
+}
+
+/* Check if the Tx to sign have the same parameters as the ones previously validated */
+bool swap_check_validity() {
```

Comment:
I think it's more clear if we take the elements to verify in parameter instead of accessing them from global
```
bool swap_check_validity(amount, destination, fees) {
```

Reply:
Ok done